packages/astro/src/core/build/plugins/plugin-ssr.ts

```diff
@@ -134,8 +134,13 @@ function buildManifest(
 		staticFiles.push(entryModules[PAGE_SCRIPT_ID]);
 	}
 
-	const bareBase = removeTrailingForwardSlash(removeLeadingForwardSlash(settings.config.base));
-	const joinBase = (pth: string) => (bareBase ? bareBase + '/' + pth : pth);
+	const prefixAssetPath = (pth: string) => {
+		if (settings.config.build.assetsPrefix) {
+			return joinPaths(settings.config.build.assetsPrefix, pth);
+		} else {
+			return prependForwardSlash(joinPaths(settings.config.base, pth));
+		}
+	};
```

Comment:
I'm not sure if this is exactly a breaking change for the SSR manifest. Before asset paths:

- `foo.css`
- `custom-base/foo.css`
- `https://example.com/_astro/foo.css`

After asset paths:

- `/foo.css`
- `/custom-base/foo.css`
- `https://example.com/_astro/foo.css`

This was changed because these raw paths are passed to 

https://github.com/withastro/astro/blob/06ddd216133fa19aa1a9ee7cf747091855730eab/packages/astro/src/core/render/ssr-element.ts#L5-L13

without `base` and `assetsPrefix` params (how SSR works). But if we let the "After" paths into the function, the `else` condition needs to check includes `://` before prepending a `/` which feels kinda hacky, so I changed this to the "Before" paths to simplify it.

Reply:
Yeah it did have the `/` for `manifest.assets` added here:

https://github.com/withastro/astro/blob/9846b3e93588626b73026b3f6499c1a62df79a24/packages/astro/src/core/build/plugins/plugin-ssr.ts#L215

but not for scripts/styles derived from here:

https://github.com/withastro/astro/blob/9846b3e93588626b73026b3f6499c1a62df79a24/packages/astro/src/core/build/plugins/plugin-ssr.ts#L165-L196

The code above are linked to `main`, which used the `joinBase` helper, and the helper strips the leading slash.